### ISSUE TITLE
Paper #1 Peer Review by Ziheng Zhong

### ISSUE THREAD

**Issue by iJustinn** [YifanTong02/Term-Paper1#1]
### Introduction

The project ‘Understanding The Toronto Shelter System Flow Pattern’ demonstrates a clear goal and a structured approach to data analysis using R. The abstract clearly stated the importance of this research, the goal of the research, and the results that have been found.

The introduction did set a good context for this research, with a comprehensive description of what a shelter is. However, some parts are just repeating what’s been said in the abstract section. I don’t think it’s a good idea to repeat yourself in two adjacent sections.

> Abstract: The paper finds that males are more likely to be homeless than females.

> Introduction: The research renders the following main findings. Firstly, males are much more likely to be homeless than females.

There is an overview of the structure for this paper but did not use any internal cross-reference in the `qmd` file, which could be added for a better reading experience.

> The paper is organized as follows. The data section presents a brief introduction on the context and characteristics of the data…

### Code Structure

Then for the aspect of coding structure, the script did include everything required: simulate, download, and cleaning. For the download script specifically, it seems you did not use the way reached on the course website, but still a valid way to download the data.

> pks <- search_packages("shelter system") # get packages
> resource <- list_package_resources(pks[1,])# get resources

Also, there is a line of comment in the download data script. That may be the line you used to remind yourself while developing the code, maybe it would be better to delete in your final submission. 

> \# [...UPDATE THIS...]
> \# change the_raw_data to whatever name you assigned when you downloaded it.
> write_csv(the_raw_data, "inputs/data/unedited_data.csv") 

Then for the data cleaning script, again, it would be better to just delete the unused lines than leave them there.

> \#### Clean data ####
> \# data <- read_csv("inputs/data/unedited_data.csv")
> data <- read_csv("unedited_data.csv")

The data was correctly processed with the use of `rename()`, `select()`, and `filter()`. Moreover, the name of the parameter used was consistent, which increased the readability of your code.

Despite all these, the descriptions of your data in the paper were good. You clearly mentioned how the data was collected and the frequency of the data collection. This shows the good effort you've been putting into the data exploration and analysis.

> The data is monthly that records the entering and leaving in the shelter system, and includes more details about demographics and previous housing status. 

> The data was last refreshed on Jan. 14 2024, and the data is of high quality in terms of freshness and completeness.

### Comments

Comments in all your scripts are short and mainly descriptive, meaning only focusing on what you want the code to do rather than the reason for doing so. Adding some in-depth comments that tell the readers why you choose specific methods/functions are critical for our research since we want everyone to be able to reproduce the project without needing to ask you anything. Take a look at those example works on the course website.

> \# convert date type from character to date
> data$`date(mmm-yy)` <- my(data$`date(mmm-yy)`)

> \# extract year
> data$Year <- format(data$date, "%Y")

For instance, when you were doing things such as `select()`, provide more description on the reason behind selecting that specific column.

>   \# select columns
>   select(date, population_group, newly_identified, actively_homeless, 
>          gender_male, gender_female, moved_to_housing) %>%

### Data Processing and Visualization

In the data processing phase, your work was reasonable but need more complexity and depth. There are only 32 lines in your clean data scripts, including `#### Preamble ####`, `#### Workspace setup ####` and `read/wtire csv files`. This is surely not ideal for such projects like this.

Given there are multiple charts in your final paper, a possible way to improve on this could be further processing your data into several separate files, in order to assist your charts creation. These processes could be further selecting used columns in specific charts, since there are 8 columns in your cleaned data set, that must not all be used for every single chart you created.

Speaking of the charts you've created, they are all good quality. I can understand what you want you express by the chart in a short amount of time. Then the choice of bar chart and trendline diagram in different sections are all good fits based on the discussion you were making. For example, a trendline diagram was shown above the paragraph:

> _Figure 3（Figure 3） investigates the time trend on the number of people who are newly identifed into the shelter system and people who have moved out of the system to a house.

### Conclusion

In summary, the project 'Understanding The Toronto Shelter System Flow Pattern' is a solid research in utilizing data analysis to investigate the important social issue. It highlights the potential of data-driven analysis in informing policy decisions and social interventions. While there are areas for improvement in terms of content organization and code documentation, these do not detract significantly from the overall quality of the work. With some modifications in your data analysis and script commenting, this project will make a significant contribution to our understanding of the dynamics within the Toronto Shelter System.
